WordPress/gutenberg
# Implement "Add block" UI for Nav block Link UI

## What?


Adds a new "Add Block" control to the Link UI which is shown when adding links in the Navigation block.

Closes https://github.com/WordPress/gutenberg/issues/50888

Closes https://github.com/WordPress/gutenberg/issues/47063

Co-authored-by: getdave <get_dave@git.wordpress.org>
Co-authored-by: scruffian <scruffian@git.wordpress.org>
Co-authored-by: ciampo <mciampini@git.wordpress.org>
Co-authored-by: stokesman <presstoke@git.wordpress.org>
Co-authored-by: draganescu <andraganescu@git.wordpress.org>
Co-authored-by: richtabor <richtabor@git.wordpress.org>
Co-authored-by: SaxonF <saxonafletcher@git.wordpress.org>

## Why?


Currently when the user clicks the plus icon within the Nav block, it defaults to auto-adding custom link variant of the `core/navigation-link` block. However this makes it hard to add other types of blocks such as search, or other variants such as Page...etc

This PR improves the UX by offering an `Add block` control add the bottom of the Link UI. When clicked the Link UI disappears and the block "Quick Inserter" is displayed with all the blocks that are allowed to be inserted.

If the user selects a block then it is inserted. If they click back then the Link UI is displayed again.

## How?


- Export `QuickInserter` component from Block Editor package (we might want to make this private for now)
- Set up internal state within Nav block Link UI to control toggling between adding a link and adding a block.


## Testing Instructions





- New Post
- Add Nav block
- Make sure Nav Menu is empty.
- Click `+` to add a new block.
- See Link UI appear.
- See `Add block` shown add bottom of Link UI.
- Click `Add block`
- See list of blocks.
- Click a block.
- See that is inserted
- Try same again but this time test that the `Back` button works.

### Testing Instructions for Keyboard


## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/444434/28022613-e5f1-42a4-b0d7-564d12f60fe5
